### DISCUSSION TITLE
DevTools "add column" button reported as not working

### DISCUSSION THREAD

**Discussion by tabacitu** [Laravel-Backpack/community-forum#354] [Q&A (Help)] [answered]
Fabiano just reported on email that "_I updated the packages, however there is now an issue with the add model in DevTools. The add column is not adding. It’s not working. What else could be wrong with?_"

![image002](https://user-images.githubusercontent.com/1032474/210783566-34e34611-0d8c-4dc6-9316-fc1e1df7a29a.png)

@pxpm can you please investigate?

**Comment by pxpm** [comment]
Sure thing @tabacitu , unfortunately I wasn't able to reproduce the issue. 

I think there must be some JS error. 

I've just freshly installed devtools, and I seen an important thing missing in the screenshot you provided, that I have on mine:
![image](https://user-images.githubusercontent.com/7188159/210790110-b78f41b5-83f3-4124-aa54-131e73074e39.png)

So when the type of column change, those inputs are dynamically created. In the provided screenshot I can see that the JS didn't trigger those changes, so in `Developer Console` the must be some JS error logged. 

Possible solutions include:
- republish the backpack assets after the `composer update` with `php artisan backpack:publish-assets`
- `Hard Reload & Clear Cache` in browser

Let me know

**Reply by fabianotrentin** [reply]
Hi @pxpm 

I have attached my screen with the Developer Console. 
Before that, I followed your suggestions to republish the backpack assets, hard reload and clear cache, but still the same problem.

![image](https://user-images.githubusercontent.com/2879612/211719553-d863f170-2d0d-4248-807c-5852a88e75d6.png)

**Reply by tabacitu** [reply]
Ugh looks like Livewire isn't installed. Right @pxpm ?

**Reply by pxpm** [reply]
Indeed @tabacitu that seems to be the case. Can you do a `composer why livewire/livewire`  @fabianotrentin ? Just to make sure if it's the package missing, or it's an error in the loading of scripts. 

Cheers

**Reply by fabianotrentin** [reply]
Hi @pxpm
The command returns the following message:
backpack/devtools 1.3.0 requires livewire/livewire (^2.4)

**Reply by pxpm** [reply, marked as answer]
@fabianotrentin there must be some issue with your setup then, and the livewire scripts are not beeing loaded. 

I see that your url looks like `localhost/**public**/admin`, when usually it should be only `localhost/admin` without the public part, because that public part should be handled by your php server using rewrite rules in the configuration. 

I think you probably need to setup a vhost (say http://test.local) that points to your `www/test/public` directory. 

Maybe this could help you: https://medium.com/@ajtech.mubasheer/configure-a-virtual-host-for-laravel-project-in-xampp-for-windows-10-d3f0068e7e1b

Also, can you check the `Network` tab on developer console and see where the request to load livewire (that returns 404) is pointing to ?

Cheers

**Reply by fabianotrentin** [reply]
@pxpm it seems to be working smoothly now. Thanks.
